pandas/core/generic.py

```diff
@@ -2795,7 +2798,10 @@ def to_hdf(
     def to_sql(
         self,
         name: str,
-        con,
+        con: sqlalchemy.engine.Engine
```

Comment:
You'll need to use `deprecate_nonkeyword_arguments` to first deprecate this

Reply:
@mroeschke Looks like the above is a decorator. 

So I would use the decorator on `to_sql`? I'm guessing that the values for the params for the decorator will be:

- version: 3.0
- allowed_args: buf/path_or_buf
- DataFrame.to_sql


Outside of updating unit tests to handle the use of the decorator, would this be the correct method to take for this PR?

Reply:
@mroeschke  Looks like we've got a bunch of failing tests in `/tests/resample`

Also, would the preferred way for me to update the unit tests be to catch the FutureWarning message via `with tm.assert_produces_warning(FutureWarning, match=re.escape(msg)):`?